src/Resources/UserSessionResource.php

```diff
@@ -201,6 +201,60 @@ protected function handleLogin(array $credentials = [], $remember = false)
         }
     }
 
+    /**
+     * @param array $credentials
+     * @return string
+    */
+    private function generateToken($credentials) {
+        $integrateio_id = $credentials['integrateio_id'];
+        $email = $credentials['email'];
+        $timestamp = $credentials['timestamp'];
+        $secret = getenv('INTEGRATEIO_SSO_SECRET');
+
+        $hashedToken = sha1($integrateio_id . ':' . $email . ':' . $secret . ':' . $timestamp);
+        return $hashedToken;
+    }
+     /**
+     * Performs login for Integrate Io Hosted trial Users.
+     *
+     * @param array $credentials
+     * @param bool  $remember
+     *
+     * @return array
+     * @throws BadRequestException
+     * @throws UnauthorizedException
+     * @throws \Exception
+     */
+    protected function handleIntegrateLogin(array $credentials = [], $remember = false)
+    {
+        // Check all params are there:
+        $requiredParams = array('integrateio_id', 'email', 'sso_token', 'timestamp');
+
+        // Make sure that all params are in the request.
+        foreach ($requiredParams as $requiredParam) {
+            if (!isset($credentials[$requiredParam])) {
+                throw new BadRequestException('Missing Parameters');
+            }
+        }
+
+        if (($this->generateToken($credentials) === $credentials['sso_token']) && ($credentials['timestamp'] > (time() - 120))) {
+            $credentials['is_active'] = 1;
```

Comment:
@tomonorman @krishnapriawan 
Can you please elaborate, what's the purpose of this `is_active`?

Reply:
Part of the laravel Login authentication. Users can be active or inactive (Rather than just deleted). Its set here to check against the db (make sure same there).

But I think you have a point. given the integrate login is a separate process this may not be necessary in this function

Reply:
Okay thanks for clarifying. Up to you whether you want to remove it.